### PR TITLE
Refactor PG and S3 code to be swappable with other implementations

### PR DESCRIPTION
**Pull request by marccampbell**
This change moves all of the S3 and Postgresql implementation into a new `S3PGStore`, that implements a new `KOTStore` interface. This interface wraps all persistence in the Admin Console, and nothing else will write to S3 and Postgres, or read from S3 and Postgres.

The goal of this is to build out the `OCIStore` that exists as an alternate interface to the `S3PGStore`. KOTS will be transitioning to a more flexible backend storage option, and the implementation details of where to store app archives, support bundles, and everything else will be handled by the store.

This store is rather large at this point. The goals with this PR was to refactor the store to the interface and stub the not implemented OCIStore. But there's more work that can be designed to simplify this store in the future.

### CODE REVIEW

**Comment by salahalsaleh** [comment]
@marccampbell yes, I highly expect that the `listApps` query will be moved to go in 1.19.

**Comment by marccampbell** [comment]
hold for 1.19

File: kotsadm/api/src/healthz/healthz_store.ts [hunk 1]
```diff
@@ -21,7 +21,14 @@ export class HealthzStore {
     const query = `select count(1)`;
     await this.pool.query(query);
 
-    const storageReady = await bucketExists(this.params, this.params.shipOutputBucket);
+    let storageReady = false;
+
+    if (this.params.storageBaseURI.startsWith("docker://")) {
+      storageReady = true;  // TODO
```
Comment (divolgin) [inline kotsadm/api/src/healthz/healthz_store.ts:27]: console.log something here?  this is bound to be forgotten.

File: kotsadm/kustomize/overlays/dev/kustomization.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 bases:
   - ../../base
-  - ./minio
+  # - ./minio
```
Comment (divolgin) [inline kotsadm/kustomize/overlays/dev/kustomization.yaml:3]: i don't see where registry is added to replace minio in dev

Comment (marccampbell) [reply kotsadm/kustomize/overlays/dev/kustomization.yaml:3]: 8 lines down

File: kotsadm/pkg/store/ocistore/supportbundle_store.go [hunk 3]
```diff
@@ -1,44 +1,54 @@
...
-	kotss3 "github.com/replicatedhq/kots/kotsadm/pkg/s3"
+	supportbundletypes "github.com/replicatedhq/kots/kotsadm/pkg/supportbundle/types"
+	troubleshootredact "github.com/replicatedhq/troubleshoot/pkg/redact"
 	"go.uber.org/zap"
 )
 
-func refFromBundleID(bundleID string, baseURI string) string {
-	baseURI = strings.TrimSuffix(baseURI, "/")
+func (s OCIStore) ListSupportBundles(appID string) ([]*supportbundletypes.SupportBundle, error) {
+	return nil, errors.New("not implemented")
```
Comment (emosbaugh) [inline kotsadm/pkg/store/ocistore/supportbundle_store.go:24]: im curious to know how this will be implemented

**Pull request merged**